Source/Particles/ShapeFactors.H

```diff
@@ -19,6 +25,13 @@ int compute_shape_factor(amrex::Real* const sx, amrex::Real xint)
     return 0;
 };
 
+template <int depos_order>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+int compute_shape_factor_dbl(double* const sx, double xint)
```

Comment:
You could probably introduce those as templates, as they are inline header functions already :)
```C++
template <int depos_order, typename T=amrex::Real>
AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
int compute_shape_factor(T* const sx, T xint)
// ...
```
called as:
```C++
compute_shape_factor<depos_order>( /* ... */ ) // Real
compute_shape_factor<depos_order, double>( /* ... */ ) // double
```

Reply:
These are now introduced as functors, like you suggested and outlined, thanks again for explaining the details!